### PR TITLE
Add support for containerd

### PR DESCRIPTION
**Pull request by ivan4th**
This PR adds support for `DIND_CRI=containerd` option which makes k-d-c use containerd as its CRI implementation.

### CODE REVIEW

File: image/docker.service [hunk 1]
```diff
@@ -1,35 +1,47 @@
...
 Wants=network-online.target
-Requires=docker.socket
 
 [Service]
-Environment=container=docker
 Type=notify
 # the default is not to use systemd for cgroups because the delegate issues still
 # exists and systemd currently does not support the cgroup feature set required
 # for containers run by docker
+#ExecStart=/usr/bin/dockerd -H unix://
```
Comment (jellonek) [inline image/docker.service]: This can be removed?

Comment (ivan4th) [reply image/docker.service]: Done.

**Pull request merged**